contracts/src/v0.1/Prepayment.sol

```diff
@@ -251,28 +278,195 @@ contract Prepayment is Ownable, IPrepayment, ITypeAndVersion {
         return sAccIdToAccount[accId].getNonce(consumer);
     }
 
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getAccountDetail(
+        uint64 accId
+    ) external view returns (uint256, uint256, uint256, uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getAccountDetail();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getSubscriptionPaid(uint64 accId) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getSubscriptionPaid();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function isValidReq(uint64 accId) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.isValidReq();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getFeeRatio(uint64 accId) external view returns (uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getFeeRatio();
+    }
+
     /**
      * @inheritdoc IPrepayment
      */
     function createAccount() external returns (uint64) {
         uint64 currentAccId = sCurrentAccId + 1;
         sCurrentAccId = currentAccId;
 
-        Account acc = new Account(currentAccId, msg.sender);
+        Account acc = new Account(currentAccId, msg.sender, IAccount.AccountType.KLAY_REGULAR);
         sAccIdToAccount[currentAccId] = acc;
 
-        emit AccountCreated(currentAccId, address(acc), msg.sender);
+        emit AccountCreated(
+            currentAccId,
+            address(acc),
+            msg.sender,
+            IAccount.AccountType.KLAY_REGULAR
+        );
         return currentAccId;
     }
 
     /**
      * @inheritdoc IPrepayment
      */
-    function createTemporaryAccount(address owner) external returns (uint64) {
+    function createFiatSubscriptionAccount(
```

Comment:
It is good to separate the creation of different account types! However, in all cases `createFiatSubscriptionAccount`, `createKlaySubscriptionAccount` and `createKlayDiscountAccount`, we have to set who is the owner of those newly generated accounts. Currently, `onlyOwner` becomes the owner of all such accounts and real owners won't be able to use it properly (e.g. adding consumers). Let's add one more parameter (`address accountOwner`) to all functions and pass it to `Account` contract and `AccountCreated` instead of `msg.sender`.

Reply:
In tests, I noticed that you expect to transfer the ownership of the account to actual user through `requestAccountOwnerTransfer` and `acceptAccountOwnerTransfer`. It is also good option. The main purpose at that time was that if somebody already has account with KLAY and setup all consumers, he does not want to lose it easily while transferring ownership. With the suggestion above, the account does not have comparable value and is a bit easier for users to get. They don't need to "accept" the ownership. Anyway, I think you can decided what you consider a better approach.